src/Chapter16.Tests/Listing16.15.Tests.cs

```diff
@@ -1,22 +1,98 @@
 ﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Diagnostics;
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter16.Listing16_15.Tests
 {
     [TestClass]
     public class ProgramTests
     {
+        public TestContext TestContext { get; set; } = null!; // Assigned by MSTest.
+
         [TestMethod]
         public void SelectingAnonymousTypeFollowingGroupClause()
         {
+            string e = Shared.NetCore.GetNetCoreVersion() switch
```

Comment:
This entire switch statement appears unused and duplicated the one below.

Reply:
Indeed... incomplete refactoring. :)

Reply:
Indeed... incomplete refactoring. :)